README.md
# React Native Leanplum

## install
```bash
npm install --save @gleeph/react-native-leanplum
react-native link @gleeph/react-native-leanplum
```

## configuration

### android

Add Leanplum keys into the AndroidManifest.xml file.

```xml
<meta-data android:name="com.reactnativeleanplum.APP_ID" android:value="insert here the leanplum APP ID" />
<meta-data android:name="com.reactnativeleanplum.DEV_KEY" android:value="insert here the leanplum Developpement key" />
<meta-data android:name="com.reactnativeleanplum.PROD_KEY" android:value="insert here the leanplum Production key" />
```
check the MainApplication.java file and replace : 

~~new RNLeanplumPackage()~~

 by
 
new RNLeanplumPackage(MainApplication.this)


add in your build.gradle
```
allprojects {
    repositories {
        ...
        maven { url "https://repo.leanplum.com/" }
        ...
    }
}
```
### ios

Add Leanplum keys into the Info.plist file.

```xml
<key>LeanplumAppId</key>
<string>insert here the leanplum APP ID</string>
<key>LeanplumDevSecret</key>
<string>insert here the leanplum Developpement key</string>
<key>LeanplumProdSecret</key>
<string>insert here the leanplum Production key</string>
```

[![npm version](https://badge.fury.io/js/%40brandingbrand%2Freact-native-leanplum.svg)](https://badge.fury.io/js/%40brandingbrand%2Freact-native-leanplum)

> README is WiP

# Install

```sh
yarn add @brandingbrand/react-native-leanplum
```

# Usage

> You may need `https://github.com/brandingbrand/react-native-push-notification-leanplum` too

## JS
```js
import Leanplum from '@brandingbrand/react-native-leanplum';
...
const leanPlumKeys = {
  appId: '',
  development: '',
  production: '',
}
...
const start = async () => {
  if (__DEV__) {
    Leanplum.setAppIdForDevelopmentMode(
      leanPlumKeys.appId,
      leanPlumKeys.development,
    );
  } else {
    Leanplum.setAppIdForProductionMode(
      leanPlumKeys.appId,
      leanPlumKeys.production,
    );
  }

  const started = await Leanplum.hasStarted();
  if (!started) Leanplum.start();
};
```

### Leanplum API

  - `setApiConnectionSettings`: <br>
  _Optional_<br>
  Sets the API server. The API path is of the form http[s]://hostname/servletName<br>
  **@param** hostName The name of the API host, such as api.leanplum.com<br>
  **@param** servletName The name of the API servlet, such as api<br>
  **@param** ssl Whether to use SSL

  - `setNetworkTimeout`: <br>
  _Optional_<br>
  Adjusts the network timeouts.
  The default timeout is 10 seconds for requests, and 15 seconds for file downloads.

  - `setNetworkActivityIndicatorEnabled`: <br>
  _iOS Only_<br>
  Sets whether to show the network activity indicator in the status bar when making requests.
  Default: YES.

  - `setCanDownloadContentMidSessionInProductionMode`: <br>
  _Advanced_<br>
  Whether new variables can be downloaded mid-session. By default, this is disabled.
  Currently, if this is enabled, new variables can only be downloaded if a push notification is sent
  while the app is running, and the notification's metadata hasn't be downloaded yet.

  - `setFileHashingEnabledInDevelopmentMode`: <br>
  Modifies the file hashing setting in development mode.
  By default, Leanplum will hash file variables to determine if they're modified and need
  to be uploaded to the server if we're running in the simulator.
  Setting this to NO will reduce startup latency in development mode, but it's possible
  that Leanplum will not always have the most up-to-date versions of your resources.

  - `setVerboseLoggingInDevelopmentMode`: <br>
  _iOS Only_<br>
  Sets whether to enable verbose logging in development mode. Default: NO.

  - `setInAppPurchaseEventName`: <br>
  _iOS Only_<br>
  Sets a custom event name for in-app purchase tracking. Default: Purchase.

  - `setAppIdForDevelopmentMode`: <br>
  Must call either this or {@link setAppIdForProductionMode}
  before issuing any calls to the API, including start.<br>
  **@param** appId Your app ID.<br>
  **@param** accessKey Your development key.

  - `setAppIdForProductionMode`: <br>
  Must call either this or {@link setAppIdForDevelopmentMode}
  before issuing any calls to the API, including start.<br>
  **@param** appId Your app ID.<br>
  **@param** accessKey Your production key.

  - `setDeviceId`: <br>
  Sets a custom device ID. For example, you may want to pass the advertising ID to do attribution.
  By default, the device ID is the identifier for vendor.

  - `setAppVersion`: <br>
  _iOS Only_<br>
  By default, Leanplum reports the version of your app using CFBundleVersion, which
  can be used for reporting and targeting on the Leanplum dashboard.
  If you wish to use CFBundleShortVersionString or any other string as the version,
  you can call this before your call to Leanplum.start()

  - `start`: <br>
  Call this when your application starts.
  This will initiate a call to Leanplum's servers to get the values
  of the variables used in your app.

  - `hasStarted`: <br>
  Returns whether or not Leanplum has finished starting.

  - `hasStartedAndRegisteredAsDeveloper`: <br>
  Returns whether or not Leanplum has finished starting and the device is registered
  as a developer.

  - `onStartResponse`: <br>
  Function to call when the start call finishes, and variables are returned
  back from the server. Calling this multiple times will call each function
  in succession.

  - `onVariablesChanged`: <br>
  Function to call when the variables receive new values from the server.
  This will be called on start, and also later on if the user is in an experiment
  that can update in realtime.

  - `onInterfaceChanged`: <br>
  _iOS Only_<br>
  Function to call when the interface receive new values from the server.
  This will be called on start, and also later on if the user is in an experiment
  that can update in realtime.

  - `onVariablesChangedAndNoDownloadsPending`: <br>
  Function to call when no more file downloads are pending (either when
  no files needed to be downloaded or all downloads have been completed).

  - `onceVariablesChangedAndNoDownloadsPending`: <br>
  Function to call ONCE when no more file downloads are pending (either when
  no files needed to be downloaded or all downloads have been completed).

  - `setUserId`: <br>
  Updates a user ID after session start.

  - `setUserAttributes`: <br>
  Updates a user ID after session start with a dictionary of user attributes.

  - `setTrafficSourceInfo`: <br>
  Sets the traffic source info for the current user.
  Keys in info must be one of: 
    - `publisherId`
    - `publisherName`
    - `publisherSubPublishe`
    - `publisherSubSite`
    - `publisherSubCampaign`
    - `publisherSubAdGroup`
    - `publisherSubAd`

  - `advanceTo`: <br>
  Advances to a particular state in your application. The string can be
  any value of your choosing, and will show up in the dashboard.
  A state is a section of your app that the user is currently in.
  You can specify up to 200 types of parameters per app across all events and state.
  The parameter keys must be strings, and values either strings or numbers.<br>
  **@param** state The name of the state. (nullable)<br>
  **@param** info Anything else you want to log with the state. For example, if the state
  is watchVideo, info could be the video ID.<br>
  **@param** params A dictionary with custom parameters.

  - `pauseState`: <br>
  Pauses the current state.
  You can use this if your game has a "pause" mode. You shouldn't call it
  when someone switches out of your app because that's done automatically.

  - `resumeState`: <br>
  Resumes the current state.

  - `trackAllAppScreens`: <br>
  Automatically tracks all of the screens in the app as states.
  You should not use this in conjunction with advanceTo as the user can only be in
  1 state at a time. This method requires LeanplumUIEditor module.

  - `trackPurchase`: <br>
  Manually track purchase event with currency code in your application. It is advised to use
  trackInAppPurchases to automatically track IAPs.

  - `trackInAppPurchases`: <br>
  _iOS Only_<br>
  Automatically tracks InApp purchase and does server side receipt validation.

  - `track`: <br>
  Logs a particular event in your application. The string can be
  any value of your choosing, and will show up in the dashboard.
  To track a purchase, use trackPurchase().

  - `variants`: <br>
  Gets a list of variants that are currently active for this user.
  Each variant is a dictionary containing an id.

  - `forceContentUpdate`: <br>
  Forces content to update from the server. If variables have changed, the
  appropriate callbacks will fire. Use sparingly as if the app is updated,
  you'll have to deal with potentially inconsistent state or user experience.
  The provided callback will always fire regardless
  of whether the variables have changed.

  - `enableTestMode`: <br>
  This should be your first statement in a unit test. This prevents
  Leanplum from communicating with the server.

  - `setTestModeEnabled`: <br>
  Used to enable or disable test mode. Test mode prevents Leanplum from
  communicating with the server. This is useful for unit tests.

  - `setPushSetup`: <br>
  _iOS Only_<br>
  Customize push setup. If this API should be called before [Leanplum start]. If this API is not
  used the default push setup from the docs will be used for "Push Ask to Ask" and
  "Register For Push".

  - `isPreLeanplumInstall`: <br>
  _iOS Only_<br>
  Returns YES if the app existed on the device more than a day previous to a version built with
  Leanplum was installed.

  - `deviceId`: <br>
  Returns the deviceId in the current Leanplum session. This should only be called after
  Leanplum.start().

  - `userId`: <br>
  Returns the userId in the current Leanplum session. This should only be called after
  [Leanplum start].

  - `inbox`: <br>
  Returns an instance to the singleton LPInbox object.

### LPInbox

### LPInboxMessage


CHANGELOG.md
# [3.1.0](https://github.com/gleephDamien/react-native-leanplum/compare/v3.0.5...v3.1.0) (2021-01-05)


### Features

* **android:** start js ([8ee9332](https://github.com/gleephDamien/react-native-leanplum/commit/8ee9332f29e135107d5bd23a7afceeb1cf3d5fd9))
* **ios:** js starting ([95a9566](https://github.com/gleephDamien/react-native-leanplum/commit/95a95669d13e69e0abd74a915a72f19f8a8731f9))

## [3.0.5](https://github.com/gleephDamien/react-native-leanplum/compare/v3.0.4...v3.0.5) (2020-09-09)


### Bug Fixes

* **android:** ensure push notification color ([1af8910](https://github.com/gleephDamien/react-native-leanplum/commit/1af891082e67fb0609f8fe580b9325660c8361cf))

## [3.0.4](https://github.com/gleephDamien/react-native-leanplum/compare/v3.0.3...v3.0.4) (2020-09-08)


### Bug Fixes

* **android:** push notification handle ([7840286](https://github.com/gleephDamien/react-native-leanplum/commit/7840286cef33061ca40942da2a1bcbc384b756d8))

## [3.0.3](https://github.com/gleephDamien/react-native-leanplum/compare/v3.0.2...v3.0.3) (2020-09-08)


### Bug Fixes

* **android:** missing devMode ([8dbe99b](https://github.com/gleephDamien/react-native-leanplum/commit/8dbe99b67bec0ff99ec25631c172320ffad04552))

## [3.0.2](https://github.com/gleephDamien/react-native-leanplum/compare/v3.0.1...v3.0.2) (2020-09-07)


### Bug Fixes

* **android:** push notification ([6e1e11c](https://github.com/gleephDamien/react-native-leanplum/commit/6e1e11ca27e86b939bda69179d97d2bc92f3f439))

## [3.0.1](https://github.com/gleephDamien/react-native-leanplum/compare/v3.0.0...v3.0.1) (2019-10-30)


### Bug Fixes

* **ios:** crash on debug ([c7d85cb](https://github.com/gleephDamien/react-native-leanplum/commit/c7d85cb1ea3bd98d827bd88c3a27f6d2b48db343))

# [3.0.0](https://github.com/gleephDamien/react-native-leanplum/compare/v2.3.1...v3.0.0) (2019-10-14)


### Bug Fixes

* update packages for security alerts ([61dd29d](https://github.com/gleephDamien/react-native-leanplum/commit/61dd29d936c9e8e668cfc3b2a9e04454c3465be6))


### chore

* update android to sdk 28 ([90caefd](https://github.com/gleephDamien/react-native-leanplum/commit/90caefddce9f2cb851c23b35828a95c8c53057c2))


### Features

* **android:** add androidX support ([4ebf3df](https://github.com/gleephDamien/react-native-leanplum/commit/4ebf3dfb7b9535d949459f4ab660daa4065efe87))


### BREAKING CHANGES

* This updates the Android SDK to 28 and updates gradle settings to enable 64 bit releases.

## [3.0.1](https://github.com/brandingbrand/react-native-leanplum/compare/v3.0.0...v3.0.1) (2019-08-06)


### Bug Fixes

* update packages for security alerts ([61dd29d](https://github.com/brandingbrand/react-native-leanplum/commit/61dd29d))

# [3.0.0](https://github.com/brandingbrand/react-native-leanplum/compare/v2.0.3...v3.0.0) (2019-07-18)


### chore

* update android to sdk 28 ([90caefd](https://github.com/brandingbrand/react-native-leanplum/commit/90caefd))


### BREAKING CHANGES

* This updates the Android SDK to 28 and updates gradle settings to enable 64 bit releases.

# [2.3.0](https://github.com/gleephDamien/react-native-leanplum/compare/v2.2.3...v2.3.0) (2019-08-14)


### Bug Fixes

* **android:** application crash on start ([95bab00](https://github.com/gleephDamien/react-native-leanplum/commit/95bab00))


### Features

* **push-notif:** adding dependencies to add push notification ([a057d5a](https://github.com/gleephDamien/react-native-leanplum/commit/a057d5a))

# [2.3.0](https://github.com/gleephDamien/react-native-leanplum/compare/v2.2.3...v2.3.0) (2019-08-14)


### Features

* **push-notif:** adding dependencies to add push notification ([a057d5a](https://github.com/gleephDamien/react-native-leanplum/commit/a057d5a))

## [2.2.3](https://github.com/gleephDamien/react-native-leanplum/compare/v2.2.2...v2.2.3) (2019-06-24)


### Bug Fixes

* **ios:** correction reload ios ([ae5022a](https://github.com/gleephDamien/react-native-leanplum/commit/ae5022a))

## [2.2.2](https://github.com/gleephDamien/react-native-leanplum/compare/v2.2.1...v2.2.2) (2019-04-29)


### Bug Fixes

* **ios:** push notif initialisation ([83ebf48](https://github.com/gleephDamien/react-native-leanplum/commit/83ebf48))

## [2.2.1](https://github.com/gleephDamien/react-native-leanplum/compare/v2.2.0...v2.2.1) (2019-04-29)


### Bug Fixes

* **android:** forgotten imports ([dcad883](https://github.com/gleephDamien/react-native-leanplum/commit/dcad883))

# [2.2.0](https://github.com/gleephDamien/react-native-leanplum/compare/v2.1.4...v2.2.0) (2019-04-29)


### Features

* **android:** adding push config for android ([77d736e](https://github.com/gleephDamien/react-native-leanplum/commit/77d736e))

## [2.1.4](https://github.com/gleephDamien/react-native-leanplum/compare/v2.1.3...v2.1.4) (2019-04-15)


### Bug Fixes

* **ios:** removing useless logs in ios ([297f621](https://github.com/gleephDamien/react-native-leanplum/commit/297f621))

## [2.1.3](https://github.com/gleephDamien/react-native-leanplum/compare/v2.1.2...v2.1.3) (2019-04-11)


### Bug Fixes

* **android:** getting app_id and keys ([98e04d0](https://github.com/gleephDamien/react-native-leanplum/commit/98e04d0))

## [2.1.2](https://github.com/gleephDamien/react-native-leanplum/compare/v2.1.1...v2.1.2) (2019-04-10)


### Bug Fixes

* **version:** update version ([82c4adf](https://github.com/gleephDamien/react-native-leanplum/commit/82c4adf))

## [2.1.1](https://github.com/gleephDamien/react-native-leanplum/compare/v2.1.0...v2.1.1) (2019-04-10)


### Bug Fixes

* **ios:** erreur de dependences ([dc62f81](https://github.com/gleephDamien/react-native-leanplum/commit/dc62f81))
* **ios:** erreur de dependences ([0a19b21](https://github.com/gleephDamien/react-native-leanplum/commit/0a19b21))

## [2.0.3](https://github.com/brandingbrand/react-native-leanplum/compare/v2.0.2...v2.0.3) (2019-01-08)


### Bug Fixes

* **android:** guard against null json value ([9e7f83d](https://github.com/brandingbrand/react-native-leanplum/commit/9e7f83d))


src/Leanplum.ts
import { NativeModules } from 'react-native';
import { Dictionary } from './Dictionary';
import { LPInbox } from './LPInbox';
import { LPPushNotif } from './LPPushNotif';

export type LeanplumAttributes = Dictionary<boolean | number | string>;

export interface LeanplumTrafficSourceInfo {
  publisherId?: string;
  publisherName?: string;
  publisherSubPublisher?: string;
  publisherSubSite?: string;
  publisherSubCampaign?: string;
  publisherSubAdGroup?: string;
  publisherSubAd?: string;
};

export interface LeanplumVariant {
  id: string;
};
const {RNLeanplum}=NativeModules;
export const Leanplum = {
  /**
   * Optional. Sets the API server. The API path is of the form http[s]://hostname/servletName
   * @param hostName The name of the API host, such as api.leanplum.com
   * @param servletName The name of the API servlet, such as api
   * @param ssl Whether to use SSL
   */
  setApiConnectionSettings: (hostName: string, servletName: string, usingSsl: boolean): void => {
    return RNLeanplum.setApiConnectionSettings(hostName, servletName, usingSsl);
  },

  /**
   * Optional. Adjusts the network timeouts.
   * The default timeout is 10 seconds for requests, and 15 seconds for file downloads.
   */
  setNetworkTimeout: (seconds: number, downloadSeconds: number): void => {
    return RNLeanplum.setNetworkTimeout(seconds, downloadSeconds);
  },

  /**
   * iOS Only.
   * Sets whether to show the network activity indicator in the status bar when making requests.
   * Default: YES.
   */
  setNetworkActivityIndicatorEnabled: (enabled: boolean): void => {
    return RNLeanplum.setNetworkActivityIndicatorEnabled(enabled);
  },

  /**
   * Advanced: Whether new variables can be downloaded mid-session. By default, this is disabled.
   * Currently, if this is enabled, new variables can only be downloaded if a push notification is sent
   * while the app is running, and the notification's metadata hasn't be downloaded yet.
   */
  setCanDownloadContentMidSessionInProductionMode: (enabled: boolean): void => {
    return RNLeanplum.setCanDownloadContentMidSessionInProductionMode(enabled);
  },

  /**
   * Modifies the file hashing setting in development mode.
   * By default, Leanplum will hash file variables to determine if they're modified and need
   * to be uploaded to the server if we're running in the simulator.
   * Setting this to NO will reduce startup latency in development mode, but it's possible
   * that Leanplum will not always have the most up-to-date versions of your resources.
   */
  setFileHashingEnabledInDevelopmentMode: (enabled: boolean): void => {
    return RNLeanplum.setFileHashingEnabledInDevelopmentMode(enabled);
  },

  /**
   * iOS Only.
   * Sets whether to enable verbose logging in development mode. Default: NO.
   */
  setVerboseLoggingInDevelopmentMode: (enabled: boolean): void => {
    return RNLeanplum.setVerboseLoggingInDevelopmentMode(enabled);
  },

  /**
   * iOS Only.
   * Sets a custom event name for in-app purchase tracking. Default: Purchase.
   */
  setInAppPurchaseEventName: (event: string): void => {
    return RNLeanplum.setInAppPurchaseEventName(event);
  },

  /**
   * Must call either this or {@link setAppIdForProductionMode}
   * before issuing any calls to the API, including start.
   * @param appId Your app ID.
   * @param accessKey Your development key.
   */
  setAppIdForDevelopmentMode: (appId: string, accessKey: string): void => {
    return RNLeanplum.setAppIdForDevelopmentMode(appId, accessKey);
  },

  /**
   * Must call either this or {@link setAppIdForDevelopmentMode}
   * before issuing any calls to the API, including start.
   * @param appId Your app ID.
   * @param accessKey Your production key.
   */
  setAppIdForProductionMode: (appId: string, accessKey: string): void => {
    return RNLeanplum.setAppIdForProductionMode(appId, accessKey);
  },

  /**
   * Sets a custom device ID. For example, you may want to pass the advertising ID to do attribution.
   * By default, the device ID is the identifier for vendor.
   */
  setDeviceId: (deviceId: string): void => {
    return RNLeanplum.setDeviceId(deviceId);
  },

  /**
   * iOS Only.
   * By default, Leanplum reports the version of your app using CFBundleVersion, which
   * can be used for reporting and targeting on the Leanplum dashboard.
   * If you wish to use CFBundleShortVersionString or any other string as the version,
   * you can call this before your call to Leanplum.start()
   */
  setAppVersion: (appVersion: string): void => {
    return RNLeanplum.setAppVersion(appVersion);
  },

  /**
   * Call this when your application starts.
   * This will initiate a call to Leanplum's servers to get the values
   * of the variables used in your app.
   */
  start: (userId?: string, attributes?: LeanplumAttributes): Promise<boolean> => {
    return RNLeanplum.start(userId, attributes);
  },

  /**
   * Returns whether or not Leanplum has finished starting.
   */
  hasStarted: (): Promise<boolean> => {
    return RNLeanplum.hasStarted();
  },

  /**
   * Returns whether or not Leanplum has finished starting and the device is registered
   * as a developer.
   */
  hasStartedAndRegisteredAsDeveloper: (): Promise<boolean> => {
    return RNLeanplum.hasStartedAndRegisteredAsDeveloper();
  },

  /**
   * Function to call when the start call finishes, and variables are returned
   * back from the server. Calling this multiple times will call each function
   * in succession.
   */
  onStartResponse: (callback: (success: boolean) => void): void => {
    return RNLeanplum.onStartResponse(callback);
  },

  /**
   * Function to call when the variables receive new values from the server.
   * This will be called on start, and also later on if the user is in an experiment
   * that can update in realtime.
   */
  onVariablesChanged: (callback: () => void): void => {
    return RNLeanplum.onVariablesChanged(callback);
  },

  /**
   * iOS Only.
   * Function to call when the interface receive new values from the server.
   * This will be called on start, and also later on if the user is in an experiment
   * that can update in realtime.
   */
  onInterfaceChanged: (callback: () => void): void => {
    return RNLeanplum.onInterfaceChanged(callback);
  },

  /**
   * Function to call when no more file downloads are pending (either when
   * no files needed to be downloaded or all downloads have been completed).
   */
  onVariablesChangedAndNoDownloadsPending: (callback: () => void): void => {
    return RNLeanplum.onceVariablesChangedAndNoDownloadsPending(callback);
  },

  /**
   * Function to call ONCE when no more file downloads are pending (either when
   * no files needed to be downloaded or all downloads have been completed).
   */
  onceVariablesChangedAndNoDownloadsPending: (callback: () => void): void => {
    return RNLeanplum.onceVariablesChangedAndNoDownloadsPending(callback);
  },

  /**
   * Updates a user ID after session start.
   */
  setUserId: (userId: string): void => {
    return RNLeanplum.setUserId(userId);
  },

  /**
   * Updates a user ID after session start with a dictionary of user attributes.
   */
  setUserAttributes: (userId: string, attributes: LeanplumAttributes): void => {
    return RNLeanplum.setUserAttributes(userId, attributes);
  },

  /**
   * Sets the traffic source info for the current user.
   * Keys in info must be one of: publisherId, publisherName, publisherSubPublisher,
   * publisherSubSite, publisherSubCampaign, publisherSubAdGroup, publisherSubAd.
   */
  setTrafficSourceInfo: (info: LeanplumTrafficSourceInfo): void => {
    return RNLeanplum.setTrafficSourceInfo(info);
  },

  /**
   * Advances to a particular state in your application. The string can be
   * any value of your choosing, and will show up in the dashboard.
   * A state is a section of your app that the user is currently in.
   * You can specify up to 200 types of parameters per app across all events and state.
   * The parameter keys must be strings, and values either strings or numbers.
   * @param state The name of the state. (nullable)
   * @param info Anything else you want to log with the state. For example, if the state
   * is watchVideo, info could be the video ID.
   * @param params A dictionary with custom parameters.
   */
  advanceTo: (state: string, info?: string, params?: LeanplumAttributes): void => {
    return RNLeanplum.advanceTo(state, info, params);
  },

  /**
   * Pauses the current state.
   * You can use this if your game has a "pause" mode. You shouldn't call it
   * when someone switches out of your app because that's done automatically.
   */
  pauseState: (): void => {
    return RNLeanplum.pauseState();
  },

  /**
   * Resumes the current state.
   */
  resumeState: (): void => {
    return RNLeanplum.resumeState();
  },

  /**
   * Automatically tracks all of the screens in the app as states.
   * You should not use this in conjunction with advanceTo as the user can only be in
   * 1 state at a time. This method requires LeanplumUIEditor module.
   */
  trackAllAppScreens: (): void => {
    return RNLeanplum.trackAllAppScreens();
  },

  /**
   * Manually track purchase event with currency code in your application. It is advised to use
   * trackInAppPurchases to automatically track IAPs.
   */
  trackPurchase: (event: string, value?: number, currencyCode?: string, params?: LeanplumAttributes): void => {
    return RNLeanplum.trackPurchase(event, value || 0, currencyCode, params);
  },

  /**
   * iOS Only.
   * Automatically tracks InApp purchase and does server side receipt validation.
   */
  trackInAppPurchases: (): void => {
    return RNLeanplum.trackInAppPurchases();
  },

  /**
   * Logs a particular event in your application. The string can be
   * any value of your choosing, and will show up in the dashboard.
   * To track a purchase, use trackPurchase().
   */
  track: (event: string, value?: number, info?: string, params?: LeanplumAttributes): void => {
    return RNLeanplum.track(event, value || 0, info, params);
  },

  /**
   * Gets a list of variants that are currently active for this user.
   * Each variant is a dictionary containing an id.
   */
  variants: (): Promise<Dictionary<LeanplumVariant>[]> => {
    return RNLeanplum.variants();
  },

  /**
   * Forces content to update from the server. If variables have changed, the
   * appropriate callbacks will fire. Use sparingly as if the app is updated,
   * you'll have to deal with potentially inconsistent state or user experience.
   * The provided callback will always fire regardless
   * of whether the variables have changed.
   */
  forceContentUpdate: (callback: () => void): void => {
    return RNLeanplum.forceContentUpdate(callback);
  },

  /**
   * This should be your first statement in a unit test. This prevents
   * Leanplum from communicating with the server.
   */
  enableTestMode: (): void => {
    return RNLeanplum.enableTestMode();
  },

  /**
   * Used to enable or disable test mode. Test mode prevents Leanplum from
   * communicating with the server. This is useful for unit tests.
   */
  setTestModeEnabled: (isTestModeEnabled: boolean) => {
    return RNLeanplum.setTestModeEnabled();
  },

  /**
   * iOS Only.
   * Customize push setup. If this API should be called before [Leanplum start]. If this API is not
   * used the default push setup from the docs will be used for "Push Ask to Ask" and
   * "Register For Push".
   */
  setPushSetup: (callback: () => void): void => {
    return RNLeanplum.setPushSetup(callback);
  },

  /**
   * iOS Only.
   * Returns YES if the app existed on the device more than a day previous to a version built with
   * Leanplum was installed.
   */
  isPreLeanplumInstall: (): Promise<boolean> => {
    return RNLeanplum.isPreLeanplumInstall();
  },

  /**
   * Returns the deviceId in the current Leanplum session. This should only be called after
   * Leanplum.start().
   */
  deviceId: (): Promise<string> => {
    return RNLeanplum.deviceId();
  },

  /**
   * Returns the userId in the current Leanplum session. This should only be called after
   * [Leanplum start].
   */
  userId: (): Promise<string> => {
    return RNLeanplum.userId();
  },

  /**
   * Returns an instance to the singleton LPInbox object.
   */
  inbox: () => {
      return LPInbox;
  },
  /**
   * Returns an instance to the singleton LPPushNotif object.
   */
  pushNotif:()=>LPPushNotif
};
